hw/ip/usbdev/data/usbdev.hjson

```diff
@@ -223,6 +223,56 @@
         }
       ]
     }
+    { multireg: {
+        name: "ep_out_enable",
+        count: "NEndpoints"
+        cname: "Endpoint"
+        desc: '''
+              Enable an endpoint to respond to transactions in the downstream
+              direction. Note that endpoint 0 must be enabled in both the IN
```

Comment:
What happens if endpoint 0 is disabled when the device is active? Is it worth to specify this. Would it be safer to just ignore the enable bits for Endpoint 0 and force it to be always enabled?

Reply:
If endpoint 0 is disabled, the device stops responding to control transactions on that endpoint, which would be a violation of the spec (and might eventually cause the host to think the device is in a bad state). This isn't any different from dropping `rxenable_setup[0]` or `rxenable_out[0]`, though.

I don't think there's a significant need to handle endpoint 0 as a special case. This is just saying that the order should be...
1. Enable endpoint 0 IN and OUT (also, mark the OUT endpoint as a control endpoint and set up the buffer pool)
2. "Enable" the USB device (i.e. Let the IP know it can connect the pull-up when VBUS is detected.)

As far as I can tell, our register tool is not capable of handling these special cases with multireg, so I'm thinking we just leave it like this, at least for now. If we find time to improve the tooling so the register layout doesn't change, we can eliminate a little logic and make all of endpoint 0 configuration read-only.

Reply:
My view is keeping this under sw control is good. Yes, there are constraints on the sw for compliance and we should spell them out. But this could equally be used by sw to emulate a badly behaved device or generate some weird condition.

Reply:
This sounds good to me. Thanks for the clarification @a-will and @mdhayter .